rs/sns/swap/src/swap.rs

```diff
@@ -1616,155 +1619,41 @@ impl Swap {
-                    let parsed_buyer_principal = match string_to_principal(buyer_principal) {
-                        Some(p) => p,
-                        // principal_str should always be parseable as a PrincipalId as that is enforced
-                        // in `refresh_buyer_tokens`. In the case of a bug due to programmer error, increment
-                        // the invalid field. This will require a manual intervention via an upgrade to correct
-                        None => {
-                            sweep_result.invalid += 1;
-                            continue;
-                        }
-                    };
-
-                    (None, parsed_buyer_principal, None)
-                }
-                Some(Investor::CommunityFund(cf_investment)) => {
-                    let nf_neuron_nns_controller = match cf_investment.try_get_controller() {
-                        Ok(controller) => controller,
-                        Err(e) => {
-                            log!(ERROR, "Invalid NF neuron: recipe={recipe:?} error={e}");
-                            sweep_result.invalid += 1;
-                            continue;
-                        }
-                    };
-                    (
-                        Some(nf_neuron_nns_controller),
-                        PrincipalId::from(nns_governance),
-                        Some(cf_investment.nns_neuron_id),
-                    )
-                }
-                // SnsNeuronRecipe.investor should always be present as it is set in `commit`.
-                // In the case of a bug due to programmer error, increment the invalid field.
-                // This will require a manual intervention via an upgrade to correct
-                None => {
-                    log!(
-                        ERROR,
-                        "Missing investor information for neuron recipe {:?}",
-                        recipe,
-                    );
-                    sweep_result.invalid += 1;
-                    continue;
-                }
-            };
-
-            let (dissolve_delay_seconds, memo, followees) = match recipe.neuron_attributes.as_ref()
-            {
-                Some(neuron_attribute) => (
-                    neuron_attribute.dissolve_delay_seconds,
-                    neuron_attribute.memo,
-                    neuron_attribute.followees.clone(),
-                ),
-                // SnsNeuronRecipe.neuron_attributes should always be present as it is set in `commit`.
-                // In the case of a bug due to programmer error, increment the invalid field.
-                // This will require a manual intervention via an upgrade to correct
-                None => {
-                    log!(
-                        ERROR,
-                        "Missing neuron_attributes information for neuron recipe {:?}",
+            match recipe.to_sns_neuron_recipe(nns_governance, sns_transaction_fee_e8s) {
```

Comment:
This statement looks a bit weird: 

1. Why do we convert a recipe to a recipe?
2. Why is there only one match case? Better to use an if statement if this is intentional.

Reply:
1. It is certainly a bit confusing. I added a comment to explain it. I also misnamed the function originally. We  are converting an SnsNeuronRecipe (a Swap concept) to an SnsNeuronRecipe (an SNS Governance concept)
2. There are two match cases